pkg/sql/importer/read_import_avro_logical_test.go

```diff
@@ -0,0 +1,340 @@
+	// and consequently write to an avro file. Each element in avroCols
+	// corresponds to a column in the CRL table we'll create.
+	avroCols := []avroLogicalInfo{
+		{
+			name:     "dt",
+			avroType: "int.date",
+			crdbType: "DATE",
+			nullable: true,
+		},
+		{
+			name:     "time_micros",
+			avroType: "long.time-micros",
+			crdbType: "TIME",
+			nullable: true,
+		},
+		{
+			name:     "time_millis",
+			avroType: "int.time-millis",
+			crdbType: "TIME(3)",
+			nullable: true,
+		},
+		{
+			name:     "ts_micros",
+			avroType: "long.timestamp-micros",
+			crdbType: "TIMESTAMP",
+			nullable: true,
+		},
+		{
+			name:     "ts_millis",
+			avroType: "long.timestamp-millis",
+			crdbType: "TIMESTAMP(3)",
+			nullable: true,
+		},
+		{
+			name:     "dt_not_null",
+			avroType: "int.date",
+			crdbType: "DATE",
+			nullable: false,
+		},
+		{
+			name:     "time_micros_not_null",
+			avroType: "long.time-micros",
+			crdbType: "TIME",
+			nullable: false,
+		},
+		{
+			name:     "time_millis_not_null",
+			avroType: "int.time-millis",
+			crdbType: "TIME(3)",
+			nullable: false,
+		},
+		{
+			name:     "ts_micros_not_null",
+			avroType: "long.timestamp-micros",
+			crdbType: "TIMESTAMP",
+			nullable: false,
+		},
+		{
+			name:     "ts_millis_not_null",
+			avroType: "long.timestamp-millis",
+			crdbType: "TIMESTAMP(3)",
+			nullable: false,
+		},
+	}
+	origTableName := "orig"
+	tableCreateStmt := createTableLogicalTypes(origTableName, avroCols)
+	sqlDB.Exec(t, tableCreateStmt)
+
+	// AVRO files store dates as int32, so to make this test roundtrippable,
+	// ensure the random data generator adds dates that can get encoded as an
+	// int32
+	for _, dtCol := range []string{"dt", "dt_not_null"} {
+		sqlDB.Exec(t, fmt.Sprintf(`ALTER TABLE %s ADD CONSTRAINT no_crazy_dates_%s CHECK (%s BETWEEN '%v' AND '%v')`, origTableName, dtCol, dtCol, pgdate.LowDate, pgdate.HighDate))
+	}
+
+	// IMPORT INTO only works if timestamps are within certain bounds; thus,
+	// only generate data within these bounds.
+	// TODO(MB): figure out these exact bounds, as tree.MinSupportTime & tree.
+	// MaxSuppportedTime make the test fail
+	maxTime := time.Date(3000, time.January, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339Nano)
+	minTime := time.Date(-2000, time.January, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339Nano)
+	for _, tsCol := range []string{"ts_micros", "ts_micros_not_null", "ts_millis", "ts_millis_not_null"} {
+		sqlDB.Exec(t, fmt.Sprintf(`ALTER TABLE %s ADD CONSTRAINT no_crazy_ts_%s CHECK (%s BETWEEN '%v' AND '%v')`, origTableName, tsCol, tsCol, minTime, maxTime))
+	}
+
+	// Try at most 5 times to populate this table, else fail.
+	rng, _ := randutil.NewTestRand()
+	success := false
+	for i := 1; i <= 5; i++ {
+		numRowsInserted, err := randgen.PopulateTableWithRandData(rng, tc.Conns[0], origTableName, 30)
+		require.NoError(t, err)
+		if numRowsInserted > 5 {
+			success = true
+			break
+		}
+	}
+	require.Equal(t, true, success)
```

Comment:
I believe most of these `require` functions have optional arguments that allow you to put messages that might make debugging easier:

```suggestion
	require.Equal(t, true, success, "failed to generate random data after 5 attempts")
```